datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -774,6 +770,23 @@ protected void preConsumerPollHook() {
     }
   }
 
+  /**
+   * Pre commit hook for all operations that need to be performed before committing offsets.
+   */
+  protected void preCommitHook() { }
+
+  /**
+   * Post commit hook for all operations that need to be performed after committing offsets.
+   * @param success Indicates whether the commit attempt was successful or not.
+   */
+  protected void postCommitHook(boolean success) {
```

Comment:
It'll be good if the `postCommitHook()` can get hold of the exception causing the commit to fail.

Reply:
Since we have multiple retries for commit, caching  the last exception should be sufficient.

Reply:
I'm not sure if there's value in propagating the Kafka exception up except for logging purposes, and the code in `commitWithRetries` already logs the exception. Do you think that there's value in propagating the last exception, @somandal?

Reply:
Right, at the moment I don't really fully see a need for the postCommitHook either, so i'm not yet clear about how the exception will help as such. I'll leave it up to you @jzakaryan to make a call on this.

Reply:
Assuming there is a set of actions that follow post commit using this hook.
true/false will give a signal of commit succeed or failed but an exception can help understand why it failed. Are you saying that there are more than 1 exception thrown? Generally Kafka should send only one commit message for a given consumer group and if this method commits more than 1 CG then I suggest returning a `Map<String, Exception>`. This gives user the most flexibility in terms of using this API.